### ISSUE TITLE
filter_grep can't work with <exclude> to filter out records

### ISSUE THREAD

**Issue by jeremywu0127** [fluent/fluentd#1677]
I want to send nginx access log to elasticsearch and filter out which status is 200, so I configure the fluentd with configuration like:
```
<source>
  @type tail
  path /data/wwwlogs/net/access.log
  pos_file /var/log/td-agent/access.log.pos
  tag nginx.access
  format /^"(?<x_forward>[^\"]+)"\s-\s(?<user>\S+)\s\[(?<time>[^\]]*)\](?:\s+"(?<method>\S+))?(?:\s+(?<url>\S+))?\s+(?<http_version>\S+)\s+(?<status>[^\"]+)\s+(?<bytes>\d+)(?:\s"(?<rfc>.*)")?\s"(?<agent>.*)"\s(?<remote_addr>\S+)\s(?<time_spent>\S+).*$/
  time_format %d/%b/%Y:%H:%M:%S %z
</source>

<filter nginx.access>
  @type record_transformer
  remove_keys user,http_version,rfc,agent,bytes
  <record>
    setting_host "#{Socket.gethostname}"
  </record>
</filter>

<filter nginx.access>
  @type grep
  <exclude>
    key status
    pattern ^2\d\d$
  </exclude>
</filter>

omit match pattern...
```
it failed and send all records including which status is 200, logs like:
```
2017-08-30 16:39:37 +0800 [warn]: parameter 'key' in <exclude>
  key status
  pattern ^2\d\d$
</exclude> is not used.
2017-08-30 16:39:37 +0800 [warn]: parameter 'pattern' in <exclude>
  key status
  pattern ^2\d\d$
```
but if I use `excludeN` instead of `<exclude>` like:
```
<filter nginx.access>
  @type grep
  exclude1 status 200
</filter>
```
it works!

so why did `<exclude>` fail while `excludeN` worked?

- td-agent version: **0.12.36**
- OS: **amazon linux 2014.09**

**Comment by repeatedly** [comment]
Because it is available since fluentd v0.12.38: https://www.fluentd.org/blog/fluentd-v0.14.19-has-been-released

**Comment by repeatedly** [comment]
If you want to use new syntax, please update bundled fluentd to latest one.
